src/main/java/gameframework/game/GameUniverseDefaultImpl.java

```diff
@@ -44,7 +45,13 @@ public synchronized void removeGameEntity(GameEntity gameEntity) {
 	public void allOneStepMoves() {
 		for (GameEntity entity : gameEntities) {
 			if (entity.isMovable()) {
-				((GameMovable) entity).oneStepMove();
+				GameMovable movable = (GameMovable) entity;
+				SpeedVector vector = movable.getSpeedVector();
+				if (getMoveBlockerChecker().moveValidation(movable, vector)) {
```

Comment:
I changed it because I wanted to add a simple way for the movable to react when it can't move. That was the best solution I found.